ports/nrf/common-hal/busio/I2C.c

```diff
@@ -79,20 +79,47 @@ void common_hal_busio_i2c_never_reset(busio_i2c_obj_t *self) {
     }
 }
 
+static nrfx_err_t _safe_twim_enable(busio_i2c_obj_t *self) {
```

Comment:
Could you also call this in `common_hal_busio_i2c_construct()` instead of having mostly duplicate code here and there?

Reply:
I did wonder about this but wanted to touch the pre-existing code as little as possible. Have made the changes